tree/ntupleutil/v7/src/RNTupleImporter.cxx

```diff
@@ -409,4 +408,7 @@ void ROOT::Experimental::RNTupleImporter::Import()
    }
    if (fProgressCallback)
       fProgressCallback->Finish(ctrZippedBytes->GetValueAsInt(), nEntries);
+
+   // Destruct helper objects derived from the model before the writer is destructed
+   ResetSchema();
```

Comment:
I think that this is better taken care of using RAII, as it will also ensure clean-up if an exception is thrown.